app/models/gem_name_reservation.rb

```diff
@@ -0,0 +1,7 @@
+class GemNameReservation < ApplicationRecord
+  validates :name, uniqueness: { case_sensitive: false }, presence: true
```

Comment:
nit: also validate that `name == name.downcase` ?

Reply:
updated